File: back/src/main/java/com/tdevilleduc/urthehero/back/model/Story.java
package com.tdevilleduc.urthehero.back.model;

import lombok.*;

import javax.persistence.*;

@Entity
@NoArgsConstructor
@RequiredArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class Story {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NonNull
    private Integer storyId;
    @NonNull
    private String title;
    @NonNull
    private Integer authorId;
    @NonNull
    private Integer firstPageId;
    @NonNull
    private String detailedText;
    @NonNull
    private String image;

    @Transient
    private Integer currentPageId;
    @Transient
    private Long numberOfReaders;



}

File: back/src/main/java/com/tdevilleduc/urthehero/back/model/Story.kt
package com.tdevilleduc.urthehero.back.model

import io.swagger.v3.oas.annotations.media.Schema
import javax.persistence.*

@Entity
data class Story(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Schema(description = "\${swagger.model.story.param.storyId}")
    var storyId: Int? = null,
    @Schema(description = "\${swagger.model.story.param.authorId}")
    var authorId: Int? = null,
    @Schema(description = "\${swagger.model.story.param.firstPageId}")
    var firstPageId: Int? = null,
    @Schema(description = "\${swagger.model.story.param.title}")
    var title: String = "",
    @Schema(description = "\${swagger.model.story.param.detailedText}")
    var detailedText: String = "",
    @Schema(description = "\${swagger.model.story.param.image}")
    var image: String = "",
    @Transient
    @Schema(description = "\${swagger.model.story.param.numberOfReaders}")
    var numberOfReaders: Long? = null,
    @Transient
    @Schema(description = "\${swagger.model.story.param.numberOfPages}")
    var numberOfPages: Long? = null

)